freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaInstanceHealthDetailsService.java

```diff
@@ -78,6 +78,26 @@ public NodeHealthDetails getInstanceHealthDetails(Stack stack, InstanceMetaData
         }
     }
 
+    public static NodeHealthDetails safeGetInstanceHealthDetails(FreeIpaInstanceHealthDetailsService service, Stack stack, InstanceMetaData instance) {
+        NodeHealthDetails nodeHealthDetails;
+        try {
+            nodeHealthDetails = service.getInstanceHealthDetails(stack, instance);
+        } catch (Exception e) {
+            LOGGER.error(String.format("Unable to check the health of FreeIPA instance: %s", instance.getInstanceId()), e);
+            nodeHealthDetails = createNodeResponseWithStatusAndIssue(instance, InstanceStatus.UNREACHABLE, e.getLocalizedMessage());
+        }
+        return nodeHealthDetails;
+    }
+
+    public static NodeHealthDetails createNodeResponseWithStatusAndIssue(InstanceMetaData instance, InstanceStatus status, String issue) {
+        NodeHealthDetails nodeResponse = new NodeHealthDetails();
+        nodeResponse.setName(instance.getDiscoveryFQDN());
+        nodeResponse.setStatus(status);
+        nodeResponse.setInstanceId(instance.getInstanceId());
+        nodeResponse.addIssue(issue);
+        return nodeResponse;
+    }
+
```

Comment:
please don't make these static, these should be accessed vie injection, so these could be also mocked during unit tests
this would also resolve this strange case where the method expects the same class as parameter

Reply:
I've mad it static because it has to call the method on itself, but it is annotated with Retryable so it has to be proxied

Reply:
Moved it to a new FreeIpaSafeInstanceHealthDetailsService